File: c/0028.0_Find_the_Index_of_the_First_Occurrence_in_a_String.c
/*
Runtime: 5 ms, faster than 6.17% of C online submissions for Find the Index of the First Occurrence in a String.
Memory Usage: 5.7 MB, less than 68.08% of C online submissions for Find the Index of the First Occurrence in a String.
*/
int strStr(char * haystack, char * needle){
    int m = strlen(needle);
    int n = strlen(haystack);
    for (int windowStart = 0; windowStart <= n - m; windowStart++) {
        for (int i = 0; i < m; i++) {
            if (needle[i] != haystack[windowStart + i]) {
                break;
            }
            if (i == m - 1) {
                return windowStart;
            }
        }
    }
    return -1;
}

File: go/0028.0_Find_the_Index_of_the_First_Occurrence_in_a_String.go
/*
Runtime: 2 ms, faster than 18.41% of Go online submissions for Find the Index of the First Occurrence in a String.
Memory Usage: 2 MB, less than 36.04% of Go online submissions for Find the Index of the First Occurrence in a String.
*/
func strStr(haystack string, needle string) int {
    m := len(needle)
    n := len(haystack)
    for windowStart := 0; windowStart <= n -m; windowStart++ {
        for i := 0; i < m; i++ {
            if needle[i] != haystack[windowStart + i] {
                break
            }
            if i == m - 1 {
                return windowStart
            }
        }
    }
    return -1
}
